charts/hivemq-platform-operator/templates/deployment.yml

```diff
@@ -157,11 +157,19 @@ spec:
             timeoutSeconds: 10
           resources:
             limits:
-              cpu: {{.Values.resources.cpu | quote}}
-              memory: {{.Values.resources.memory | quote}}
+              cpu: {{ include "hivemq-platform-operator.resources-limits-cpu" . }}
```

Comment:
Can we make CPU limits optional with overrideLimits? There are some suggestions that this may help some use cases: https://kodekloud.com/blog/optimizing-clusters-for-cost-performance-part-2-impact-of-cpu-limits/#:~:text=Tim%20Hockin%2C%20who%20is%20considered,to%20equal%20to%20memory%20requests. 

So if we leave away overrideLimits.cpu then it is not set at all?

Reply:
Currently, we always set both CPU and memory limits and requests to the default values we have, ie (operator values):

```yaml
resources:
  cpu: 1024m
  memory: 1Gi
```

So if the `overrideLimits.cpu` is not set at all, we are falling back to the default value used in these above original values (i.e. resources.cpu: 1024m).
In order to not set the CPU limits at all, we would then need to modify these new overrideLimits values and explicitly disable it somehow?
```yaml
overrideLimits:
  cpu: 
    enabled: false
```
or enabled:
```yaml
overrideLimits:
  cpu: 
    enabled: true
    value: 1024m
```

Reply:
As discussed, `overrideLimits` is now modified and it accepts YAML snippets.